modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/rebalance/RebalanceRaftGroupEventsListener.java

```diff
@@ -491,24 +491,21 @@ static void doStableKeySwitch(
                 Condition con5;
                 if (plannedEntry.value() != null) {
                     // eq(revision(partition.assignments.planned), plannedEntry.revision)
-                    con5 = and(
-                            revision(plannedPartAssignmentsKey).eq(plannedEntry.revision()),
-                            value(pendingPartAssignmentsKey).eq(stableFromRaftByteArray)
-                    );
+                    con5 = revision(plannedPartAssignmentsKey).eq(plannedEntry.revision());
```

Comment:
To the reviewer: these changes fix bugs in the method, where a comparison wouldn't work if pending assignment is forced, but stable assignment is not.
In fact, this condition was not only wrong, but excessive, because we control for it in `con2`.